engine/graphics/src/graphics_private.h

```diff
@@ -167,6 +167,51 @@ namespace dmGraphics
         uint8_t m_StageFlags;
     };
 
+    struct Program
+    {
+        ProgramResourceBinding m_ResourceBindings[MAX_SET_COUNT][MAX_BINDINGS_PER_SET_COUNT];
+        dmArray<Uniform>       m_Uniforms;
+        uint8_t                m_MaxSet;
+        uint8_t                m_MaxBinding;
+    };
```

Comment:
This is the shared data needed for all programs, which is a first step to separating the graphics implementations into a "base" set of structs that are just data and an API-specific set of structs and functions that are API specific. A todo for later is to add this into the dmsdk:
`typedef struct Program* HProgram`

so that we don't need to cast it to a Program* in the functions that use it